app/helpers/apa_helper.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module ApaHelper
+  def apa_citation(article)
+    title = article.title.capitalize_all
+
+    "#{authors_in_apa_format(article)}. #{title}. #{article.journal}, #{article.volume}(#{article.issue}), \
+    #{article.first_page}-#{article.last_page}"
+  end
+
+  def authors_in_apa_format(article)
+    apa_authors = []
+
+    article.authors.split(',').each do |author|
+      last_name = author.split(' ')[0].capitalize_all
+      first_name = author.split(' ')[1..-1].map! { |a| a[0] }.join('. ')
+
+      apa_authors << ["#{last_name}, #{first_name}"]
```

Comment:
Reviewer'lara not, burada şunu yapıyorum:

"AHMET ŞAHİN, MEHMET DOĞAN, MURAT SERÇE" gibi yazar isimlerini içeren string'i APA stili yazar listesine ("Şahin, A. Doğan, M., Serçe, M.") çeviriyorum.

Daha temiz yolları mutlaka vardır, şuan en kolayıma geleni yaptım.

Reply:
Kendi koduna review atan @msdundar :)

Reply:
Eğer `authors = ['Ad Soyad', 'Ad Soyad' ]` şeklinde ise `first_name` ve `last_name` yanlış olmuş oluyor. Çünkü boşluklara göre böldüğümüzde birinci eleman Ad, diğer elemanlar soyad oluyor.

Bu göre de kodu şöyle değiştirebiliriz;

```ruby
article.authors.split(',').map do |author|
  first_name = author.split(' ')[0].capitalize_all
  last_name = author.split(' ')[1..-1].map(&:first).join('. ')
  "#{last_name}, #{first_name}"
end.join(' & ')
```

Reply:
Ben ornegi yanlis vermisim yazar listesi Soyad Ad geliyor. Denendi bu kod dogru calisiyor.